invenio_rdm_records/resources/resources.py

```diff
@@ -100,3 +100,48 @@ def search(self):
             identity=g.identity,
         )
         return items.to_dict(), 200
+
+class RDMPidProviderResource(RecordResource):
+
+
+    def create_url_rules(self):
+        """Create the URL rules for the record resource."""
+
+        def p(route):
+            """Prefix a route with the URL prefix."""
+            return f"{self.config.url_prefix}{route}"
+
+        routes = self.config.routes
+        return [
+            route("GET", p(routes["item"]), self.reserve),
+            route("DELETE", p(routes["item"]), self.unreserve),
+        ]
+
+
+
+    @request_view_args
+    @request_data
+    @response_handler()
+    def reserve(self):
+        """Reserve doi """
+        item = self.service.reseve_doi(
+            id_=resource_requestctx.view_args["pid_value"],
+            identity=g.identity,
+            data=resource_requestctx.data,
+        )
+
+        return item.to_dict(), 201
+
+
+    @request_view_args
+    def unreserve(self):
```

Comment:
question: Is it "unreserve" or does it "delete" it? My question is about workflow, and therefore naming. Since the HTTP verb used here is `DELETE` if the PID gets deleted from the local PIDStore, maybe "delete" is better name? Or "unregister" since it's what will be done at DataCite level.

Reply:
After discussions, it is probably a good idea in the end to have the URL in this format:
`/api/records/:id/draft/pids/:pid`
Then, use HTTP verbs for the action of reserving or removing.